specs/commandline/plugin.md

```diff
@@ -41,51 +42,145 @@ Aliases:
 ### notation plugin install
 
 ```text
-Installs a plugin
+Install a plugin
 
 Usage:
-  notation plugin install [flags] <plugin path>
+  notation plugin install [flags] <plugin_source>
 
 Flags:
   -h, --help                    help for install
   -f, --force                   force the installation of a plugin
-
+      --checksum string         must match SHA256 of the plugin source
+      --source string           the location of plugin installation file, options: "file", "url","registry" (default "file")                  
```

Comment:
It is possible that the type of source can be detected automatically, so that we don't need specify the option "file", "url", "registry". It is not a common practice.
- filepath
- http URL
- repo URL

A [guideline ](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md#argument-naming-conventions) for your reference

Reply:
I was hoping to omit the plugin source options either and make the source type detect by Notation automatically.  @Two-Hearts mentioned there is a special case that if a file path is as same as a repo URL, it will be hard for Notation to distinguish that case.

Reply:
> @yizha1 I was hoping to omit the plugin source options either and make the source type detect by Notation automatically. @Two-Hearts mentioned there is a special case that if a file path is as same as a repo URL, it will be hard for Notation to distinguish that case.

To correct, a file path can be as same as a valid oci artifact reference in a registry. If we are going to support oci registry in the future, then we should consider this situation in the initial design. @yizha1 @FeynmanZhou

Reply:
@FeynmanZhou @Two-Hearts Maybe we can refer to `helm install` command, which supports installation from a file/url/repo

Reply:
@yizha1 That's a good idea, from helm, I could see:
`By OCI registries: helm install mynginx --version 1.2.3 oci://example.com/charts/nginx`
They ask user to add a prefix `oci://` to the input and they explicitly check for this prefix, if it exists, then it's by oci registry. @FeynmanZhou What do you think?

Reply:
Another option might be:
1. We check the user input in the file system first. If it's a valid, existed, file path. Then we take the input as a file path and try install from there.
2. If it does not exist in the file system, then we check if it's a valid URL. If yes, install from the URL.
3. For the current iteration, if not file path nor URL, we will return an error. For future iteration, if we support oci registry, then we treat the input as an oci reference. If it's a valid reference, then we install from oci registry.

Reply:
Helm also supports installation from a repo, how does helm distinguish it is a file or a repo? For example, 

```
$ helm repo ls
NAME                    URL
harbor                  https://helm.goharbor.io

$ helm install myharbor harbor/harbor

$ ls
<no directory harbor/harbor>
```

Reply:
> Helm also supports installation from a repo, how does helm distinguish it is a file or a repo? For example,
> 
> ```
> $ helm repo ls
> NAME                    URL
> harbor                  https://helm.goharbor.io
> 
> $ helm install myharbor harbor/harbor
> 
> $ ls harbor/harbor
> ls: cannot access 'harbor/harbor': No such file or directory
> ```

@yizha1 That's a good question. The answer is, helm treats `harbor/harbor` as a file path first, they check if this is a valid file path. If the path does not exist, then they take it as a repo. So the process looks automatically on the external user's point of view.

Reply:
> @Two-Hearts If the edge case can be covered by this option, it sounds good to me.

Yes, the edge case can be covered. No matter what user input it is, Notation always check by order: file path -> URL [-> OCI reference, in the future].